python/cudf/cudf/core/dataframe.py

```diff
@@ -1109,11 +1111,15 @@ def columns(self):
         if hasattr(self, "multi_cols"):
             return self.multi_cols
         else:
-            return pd.Index(self._cols)
+            if hasattr(self, "_columns_name"):
```

Comment:
Don't we always have this attr, it's just initialized to `None` by default. This will return True, then we set `name = None`. Can just always do `name = self._columns_name`

Reply:
Done